### PR TITLE
Camil fix data transfer

### PR DESCRIPTION
**Pull request by ConfettiRaindrop**

### CODE REVIEW

File: app/src/main/java/com/example/myapplication/FirstFragment.java [hunk 1]
```diff
@@ -24,14 +24,12 @@
 
 public class FirstFragment extends Fragment {
     private FragmentFirstBinding binding;
-    //fdsafdagdsa
```
Comment (carlosfigueira) [inline app/src/main/java/com/example/myapplication/FirstFragment.java:27]: Thanks for deleting those :-)

File: app/src/main/java/com/example/myapplication/UserModel.java [hunk 2]
```diff
@@ -37,4 +50,21 @@ void doAction() throws IOException {
         osw.close();
         fos.close();
     }
+
+    public static void createFile(String fileName){
```
Comment (carlosfigueira) [inline app/src/main/java/com/example/myapplication/UserModel.java:54]: This doesn't seem to be related to the transfer issue - doesn't need to be in this PR

Comment (ConfettiRaindrop) [reply app/src/main/java/com/example/myapplication/UserModel.java:54]: this was when I was working on the file write - since it's on the same page as changes related to this PR, how would I not add it to this one? Do I have to make separate branches ahead of time?

File: app/src/main/java/com/example/myapplication/HomePage.java [hunk 3]
```diff
@@ -78,13 +81,24 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
             public void onClick(View v) {
                 NavHostFragment.findNavController(HomePage.this)
                         .navigate(R.id.action_HomePage_to_pitFragment);
             }
-        });*/
+        });
+        binding.nextInfoButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                UserModel.getMatchData().setBroke(true);
```
Comment (carlosfigueira) [inline app/src/main/java/com/example/myapplication/HomePage.java:99]: This looks like a test place to update the 'broke' status; consider updating the actual place where the 'broke' status is stored (currently in SecondFragment.java)

File: app/src/main/java/com/example/myapplication/FirstFragment.java [hunk 4]
```diff
@@ -97,6 +95,7 @@ public void onTextChanged(CharSequence s, int start, int before, int count) {
             @Override
             public void afterTextChanged(Editable s) {
                 MainActivity.teamNumber = binding.input.getText() + "";
+                UserModel.getMatchData().setTeamNumber(Integer.parseInt(MainActivity.teamNumber));
```
Comment (carlosfigueira) [inline app/src/main/java/com/example/myapplication/FirstFragment.java:98]: This may throw an exception if the value entered is not a valid number - we should validate (or try...catching) it.

File: app/src/main/java/com/example/myapplication/MatchData.java [hunk 5]
```diff
@@ -142,24 +146,64 @@ public void setDump(int dump) {
...
+        } else {
+            this.noteAcquired = NoteAcquisition.None;
+        }
+    }
 
     //Endgame
 
     public Chain getChaining() {    return chaining;    }
-    public void setChaining(Chain chaining) {    this.chaining = chaining;    }
+    public void setChaining(int index) {
```
Comment (carlosfigueira) [inline app/src/main/java/com/example/myapplication/MatchData.java:164]: Why changing from the enum value to the integer value? It will be hard to use this method

File: app/src/main/java/com/example/myapplication/PitFragment.java [hunk 6]
```diff
@@ -96,19 +95,11 @@ public void onClick(View view) {
         binding.checkBox.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
+               System.out.println(UserModel.getMatchData().isBroke());
```
Comment (carlosfigueira) [inline app/src/main/java/com/example/myapplication/PitFragment.java:98]: Test code, can be removed

File: app/src/main/java/com/example/myapplication/PitFragment.java [hunk 7]
```diff
@@ -1,5 +1,8 @@
 package com.example.myapplication;
 
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
```
Comment (carlosfigueira) [inline app/src/main/java/com/example/myapplication/PitFragment.java:5]: Are those needed?

File: app/src/main/java/com/example/myapplication/SecondFragment.java [hunk 8]
```diff
@@ -210,6 +212,7 @@ public void onClick(View view) {
             @Override
             public void onClick(View view) {
                 MainActivity.amp ++;
+                UserModel.getMatchData().setAmpTeleop(MainActivity.amp);
```
Comment (carlosfigueira) [inline app/src/main/java/com/example/myapplication/SecondFragment.java:215]: If we are storing ampTeleop in the MatchData class we should remove it from MainActivity. Here it can be something like

    MatchData matchData = UserModel.getMatchData();
    matchData.setAmpTeleop(matchData.getAmpTeleop() + 1);

Comment (carlosfigueira) [reply app/src/main/java/com/example/myapplication/SecondFragment.java:215]: Same comment for other changes in this file

File: app/src/main/java/com/example/myapplication/SecondFragment2.java [hunk 9]
```diff
@@ -133,7 +133,7 @@ public void afterTextChanged(Editable s) {
                     binding.textInput.setText(text);
                 }
                 binding.characterLimit.setText("Character Limit: " + binding.textInput.getText().length() + "/150");
-                MainActivity.teleOpNotes = binding.textInput.getText() + "";
+                UserModel.getMatchData().setNotes(String.format("%s", binding.textInput.getText()));
```
Comment (carlosfigueira) [inline app/src/main/java/com/example/myapplication/SecondFragment2.java:136]: Instead of format("%s", ...), I believe we can use `binding.textInput.getText().toString()`.

File: app/src/main/java/com/example/myapplication/SecondFragment2.java [hunk 10]
```diff
@@ -311,6 +312,7 @@ public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
             public void onClick(View v) {
                 if (MainActivity.notesThrown > 0) {
                     MainActivity.notesThrown--;
```
Comment (carlosfigueira) [inline app/src/main/java/com/example/myapplication/SecondFragment2.java:314]: Similar comment as previous file - no need to store data in MainActivity anymore:

    MatchData matchData = UserModel.getMatchData();
    matchData.setHumanPlayerNotesThrown(matchData.getHumanPlayerNotesThrown() - 1);

Comment (carlosfigueira) [reply app/src/main/java/com/example/myapplication/SecondFragment2.java:314]: Same for subsequent changes

File: app/src/main/java/com/example/myapplication/UserModel.java (hunk 2 above)
Comment (carlosfigueira) [reply app/src/main/java/com/example/myapplication/UserModel.java:54]: Once you find out that you are going to split your work in two parts (part 1: user model / match data; part 2: files), you can create a new branch from your existing branch (e.g., CamilaFileSave). In that new branch, this code would remain, while on the other branch you can delete it.

File: app/src/main/java/com/example/myapplication/FirstFragment.java (hunk 4 above)
Comment (carlosfigueira) [reply app/src/main/java/com/example/myapplication/FirstFragment.java:98]: Nevermind, the widget is taking care of it

File: app/src/main/java/com/example/myapplication/PitFragment.java [hunk 11]
```diff
@@ -77,16 +69,16 @@ public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
...
         return binding.getRoot();
 
-        //return inflater.inflate(R.layout.fragment_pit, container, false);
     }
 
     public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
 
+        ViewModelProvider viewModelProvider = new ViewModelProvider(requireActivity());
+        UserModel userModel = viewModelProvider.get(UserModel.class);
```
Comment (carlosfigueira) [inline app/src/main/java/com/example/myapplication/PitFragment.java:78]: Unused, can be removed
